### PR TITLE
Remove redundant information in tablets external compaction entry

### LINKED ISSUE #2110: Remove redundant information in tablets external compaction entry.
When an external compaction is running tablets record this in the metadata table w/ information about the external compaction.  The following is an example of this.  In the following json the fields `tmp` and `dest` are a bit redundant, could look into only having one of the fields.  All that is needed for the running external compaction is `tmp` field, that is where its writing to.  The `dest` field is what the file will be renamed to when the compaction commits.

```
4< ecomp:ECID:04d6f4a3-3caa-46e6-9ea5-ce77a0ddc8dc []	{"inputs":["hdfs://localhost:8020/accumulo/tables/4/default_tablet/A000005v.rf"],"nextFiles":[],"tmp":"hdfs://localhost:8020/accumulo/tables/4/default_tablet/A000005y.rf_tmp","dest":"hdfs://localhost:8020/accumulo/tables/4/default_tablet/A000005y.rf","compactor":"ip-10-113-12-25:9101","kind":"USER","executorId":"DCQ1","priority":288230376151711745,"propDels":false,"selectedAll":true,"compactionId":2}
```

### PR DESCRIPTION
**Pull request by dlmarion**
Compute the name of the newly compacted file from the tmp file in
DatafileManager.bringMajorCompactionOnline instead of pre-computing
it, passing it around, and storing it in the external metadata entry

Closes #2110

### CODE REVIEW

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/DatafileManager.java [hunk 1]
```diff
@@ -398,13 +398,19 @@ StoredTabletFile bringMinorCompactionOnline(TabletFile tmpDatafile, TabletFile n
...
+      DataFileValue dfv, Optional<ExternalCompactionId> ecid) throws IOException {
     final KeyExtent extent = tablet.getExtent();
     VolumeManager vm = tablet.getTabletServer().getContext().getVolumeManager();
     long t1, t2;
 
+    int idx = tmpDatafile.getMetaInsert().indexOf("_tmp");
+    String newFilePath = tmpDatafile.getMetaInsert();
+    if (idx > 0) {
+      newFilePath = newFilePath.substring(0, idx);
+    }
```
Comment (keith-turner) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/DatafileManager.java:411]: Should probably throw an exception in an else if `_tmp` suffix is not there, its expected to be there.

File: core/src/test/java/org/apache/accumulo/core/metadata/schema/TabletFileTest.java [hunk 2]
```diff
@@ -40,6 +41,16 @@ private TabletFile test(String metadataEntry, String volume, String tableId, Str
     return tabletFile;
   }
 
+  @Test
+  public void testEquivalence() {
```
Comment (keith-turner) [inline core/src/test/java/org/apache/accumulo/core/metadata/schema/TabletFileTest.java:45]: Could have a static function in DataFileManager that does the conversion and then have a unit test calls that function and verify the result.

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java [hunk 3]
```diff
@@ -973,14 +973,15 @@ public ExternalCompactionJob reserveExternalCompaction(CompactionServiceId servi
       Map<String,String> overrides =
           CompactableUtils.getOverrides(job.getKind(), tablet, cInfo.localHelper, job.getFiles());
 
-      var newFile = tablet.getNextMapFilename(!cInfo.propagateDeletes ? "A" : "C");
-      var compactTmpName = new TabletFile(new Path(newFile.getMetaInsert() + "_tmp"));
+      String tmpFileName =
+          tablet.getNextMapFilename(!cInfo.propagateDeletes ? "A" : "C").getMetaInsert() + "_tmp";
+      TabletFile compactTmpName = new TabletFile(new Path(tmpFileName));
```
Comment (keith-turner) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java:978]: Could have a little static function in CompactableUtils that does this, then call that function here and in CompactableUtils.  That makes it easier for anyone looking at the code to find the two places where this happens.

Comment (dlmarion) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableImpl.java]: Moved to a utility function in Tablet in 9d1373b

File: core/src/test/java/org/apache/accumulo/core/metadata/schema/TabletFileTest.java (hunk 2 above)
Comment (dlmarion) [reply core/src/test/java/org/apache/accumulo/core/metadata/schema/TabletFileTest.java]: Created a utility function for this and moved the test to a better location in 9d1373b

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/DatafileManager.java (hunk 1 above)
Comment (dlmarion) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/DatafileManager.java]: Added an exception in 9d1373b

**Review by keith-turner — APPROVED** [review]

File: server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableUtils.java [hunk 4]
```diff
@@ -650,4 +649,17 @@ public static MajorCompactionReason from(CompactionKind ck) {
...
   }
+
+  public static TabletFile computeCompactionFileDest(TabletFile tmpFile) {
+    String newFilePath = tmpFile.getMetaInsert();
+    int idx = newFilePath.indexOf("_tmp");
+    if (idx > 0) {
+      newFilePath = newFilePath.substring(0, idx);
+    } else {
+      throw new RuntimeException(
+          "Expected compaction tmp file " + tmpFile.getMetaInsert() + " to have suffix '_tmp'");
```
Comment (ctubbsii) [inline server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableUtils.java:660]: Is there a more specific RTE that can be used? IllegalArgumentException might be appropriate, based on this error message.

Comment (dlmarion) [reply server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/CompactableUtils.java]: changed in f75c2a8e4804685effc217c680e8cb03b852768a

**Pull request merged**